Core/SubterfugeCore/Core/Game.cs

```diff
@@ -87,5 +89,41 @@ public void LoadGameEvents(List<GameEventModel> gameEvents)
                 .ConvertAll<GameEvent>(m => GameEventFactory.parseGameEvent(m))
                 .ForEach( parsedEvent => TimeMachine.AddEvent(parsedEvent) );
         }
+
+        /// <summary>
+        /// Determines if the game is in a game over state. In Mining, ties are broken by
+        /// whoever happens to be first in the list of players.
+        /// </summary>
+        /// <returns>null if the game is not over, or the Player who has won if it is over.</returns>
+        public Player IsGameOver()
+        {
+            switch (GameMode)
+            {
+                case GameMode.MINING:
+                    foreach (Player p in TimeMachine.GetState().GetPlayers())
+                    {
+                        if (!p.IsEliminated() && p.GetNeptunium() >= Constants.MINING_NEPTUNIUM_TO_WIN)
+                        {
+                            return p;
+                        }
+                    }
+                    return null;
+
+				case GameMode.DOMINATION:
+                    foreach (Player p in TimeMachine.GetState().GetPlayers())
+                    {
+                        if (!p.IsEliminated() && TimeMachine.GetState().GetPlayerOutposts(p).Count > TimeMachine.GetState().GetOutposts().Count)
```

Comment:
This will never be true. A player can't have more outposts than the time machine has in total. Maybe for domination we this to be a win if a player has more than the sum of all other players combined? That doesn't work too much in lower player count games though